File: binaryGap.py
def solution(N):
    # write your code in Python 3.6
    if(N < 1):
        return 0
    else:
        binaryEquivalent = "{0:b}".format(N)
        strBinary = str(binaryEquivalent)
        
        savedGap = []
        countGap = 0
        start = 0
        for i in strBinary:
            if(start == 0):
                start = int(i)
            elif(start == 1 and int(i) == 0):
                countGap += 1
            elif(start == 1 and int(i) == 1):
                start = 1
                savedGap.append(countGap)
                countGap = 0
        if(savedGap==[]):
            binaryGap = 0
        else:
            binaryGap = max(savedGap)
        return binaryGap

print(solution(328))

File: javascript/binaryGap.js
solution = (N) => {
    // write your code in ES6
    if(N < 1){
        return 0;
    }else{
        let strNum = N.toString();
        let strBinary = (+strNum).toString(2);
        savedGap = [];
        countGap = 0;
        start = 0;
        for(i of strBinary){
            if(start==0){
                start = parseInt(i);
            }else if(start == 1 && parseInt(i) == 0){
                countGap +=1;
            }else if(start == 1 && parseInt(i) == 1){
                start = 1;
                savedGap.push(countGap);
                countGap = 0;
            }
        }
        if(savedGap.length < 1) return 0;
        return Math.max(...savedGap);
    }
}

console.log(solution(32))